src/vault.ts

```diff
@@ -0,0 +1,83 @@
+import { Batch } from '@mparticle/event-models';
+import { Dictionary, isEmptyObject } from './utils';
+
+export default class Vault {
+    // TODO: Make this generic
+    public contents: Dictionary<Batch>;
+
+    constructor(public key: string) {
+        this.key = key;
+        this.contents = this.getItems();
+    }
+
+    /**
+     * Stores Batches using `source_request_id` as an index
+     * @method storeBatches
+     * @param {Batches[]} batches an Array of Batches
+     */
+    storeBatches(batches: Batch[]): void {
+        this.contents = this.getItems() || {};
+
+        batches.forEach(batch => {
+            if (batch.source_request_id) {
+                this.contents[batch.source_request_id] = batch;
+            }
+        });
+
+        this.saveItems(this.contents);
+    }
+
+    /**
+     * Removes a single batch based on `source_request_id`
+     * @method removeBatch
+     * @param {String} source_request_id
+     */
+    removeBatch(source_request_id: string): void {
+        this.contents = this.getItems() || {};
+
+        delete this.contents[source_request_id];
+
+        this.saveItems<Batch>(this.contents);
+    }
+
+    /**
+     * Retrieves all batches from local storage as an array
+     * @method retrieveBatches
+     * @returns {Batch[]} a list of Batches
+     */
+    retrieveBatches(): Batch[] {
```

Comment:
*suggestion* can add modifiers here `public/private` to be explicit about the expected usages

Reply:
Good call out